cmd/mdatagen/templates/telemetry.go.tmpl

```diff
@@ -117,14 +120,15 @@ func NewTelemetryBuilder(settings component.TelemetrySettings, options ...teleme
         {{- if eq $metric.Data.Type "Histogram" -}}
         {{ if $metric.Data.Boundaries -}}metric.WithExplicitBucketBoundaries([]float64{ {{- range $metric.Data.Boundaries }} {{.}}, {{- end }} }...),{{- end }}
         {{- end }}
-        {{ if $metric.Data.Async -}}
-        metric.With{{ casesTitle $metric.Data.BasicType }}Callback(func(_ context.Context, o metric.{{ casesTitle $metric.Data.BasicType }}Observer) error {
-            o.Observe(builder.observe{{ $name.Render }}(), metric.WithAttributeSet(builder.attributeSet))
-            return nil
-        }),
-        {{- end }}
     )
     errs = errors.Join(errs, err)
+    {{- if $metric.Data.Async }}
+    builder.meter.RegisterCallback(func(_ context.Context, o metric.Observer) error {
+        o.Observe{{ casesTitle $metric.Data.BasicType }}(builder.{{ $name.Render }}, builder.observe{{ $name.Render }}(), metric.WithAttributeSet(builder.attributeSet))
+        return nil
+    }, builder.{{ $name.Render }})
+    errs = errors.Join(errs, err)
```

Comment:
This statement is not needed as nothing is changing err in the callback registration correct?

Reply:
Oh, I forgot to set the err from the registration. Fixed